CONTRIBUTING.rst

```diff
@@ -0,0 +1,310 @@
+to chime in and avoids disagreements how to progress.
+
+-  The title should summarise the issue
+-  Describe the issue in as much detail as possible in the comment
+
+GitHub does not allow editing labels, assignees or setting milestone to
+non-members of a project with at least "Triage" permission. These will have to
+be set by members with Triage permission after an issue/PR is created.
+Guidelines regarding labels, assignees and milestone therefore only concern
+members of acts-project with the necessary rights and can be ignored by others.
+
+-  Assign to yourself or leave empty
+-  Choose labels as appropriate
+
+   -  Type of issue
+   -  Which component is affected
+   -  Urgency
+   -  Fix versions
+
+-  Bug reports
+
+   -  Mention affected version(s)
+   -  Issue type: ``Bug``
+   -  A detailed description of the bug including a recipe on how to
+      reproduce it and any hints which may help diagnosing the problem
+
+-  Feature requests
+
+   -  Issue type: ``Improvement`` or ``Feature``
+   -  A detailed description of the feature request including possible
+      use cases and benefits for other users
+
+Make a contribution
+-------------------
+
+Anyone is welcome to contribute to Acts. Below is a short description how to
+contribute. If you have any questions, feel free to ask `acts-developers@cern
+<mailto:acts-developers@cern.ch>`_ for help or guidance.
+
+Please always fork the Acts repository you want to work on and create branches
+only in your own fork. Once you want to share your work, create a Pull Request
+(PR) (for gitlab users: equivalent to merge request) to the master branch of the
+upstream acts-project repository. If it is not yet ready to be merged in, please
+create a draft pull request (by clicking on the small arrow on the green "create
+pull request" button) to mark it work in progress. Once you want your branch to
+be merged in, request a review from the `reviewers team
+<https://github.com/orgs/acts-project/teams/reviewers>`_. Once a draft merge
+request is reviewed, it can be merged in.
+
+To get started with git, please refer to the `short introduction
+<http://git-scm.com/docs/gittutorial>`_ as well as the `full git documentation
+<https://git-scm.com/doc>`_. Tutorials as well as explanations of concepts and
+workflows with git can also be found on `Atlassian
+<http://www.atlassian.com/git/>`_.
+
+Checklist for pull requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+-  Your branch has been rebased on the target branch and can be
+   integrated through a fast-forward merge.
+-  A detailed description of the pull request is provided.
+-  The issue the PR closes is linked.
+-  All CI jobs pass.
+-  All newly introduced functions and classes have been documented
+   properly with doxygen.
+-  Unit tests are provided for new functionalities.
+-  For bugfixes: a test case has been added to avoid the re-appearance
+   of this bug in the future.
```

Comment:
this should reference the checklist in the pull_request_template or contain the same content.